packages/build-tools/src/generic.ts

```diff
@@ -0,0 +1,49 @@
+import path from 'path';
+
+import { BuildPhase, Generic } from '@expo/eas-build-job';
+import { BuildConfigParser, BuildStepGlobalContext, errors } from '@expo/steps';
+import nullthrows from 'nullthrows';
+
+import { BuildContext } from './context';
+import { prepareProjectSourcesAsync } from './common/projectSources';
+import { getEasFunctions } from './steps/easFunctions';
+import { CustomBuildContext } from './customBuildContext';
+import { getEasFunctionGroups } from './steps/easFunctionGroups';
+
+export async function runGenericJobAsync(ctx: BuildContext<Generic.Job>): Promise<void> {
```

Comment:
Could you briefly sum up what's the difference between this function and `runCustomBuild`?

Reply:
- no `resolveEnvFromBuildProfileAsync` (even if we wanted to keep it, we don't have "job profiles" yet, so it'd be a todo)
- `configPath` doesn't use `getReactNativeDirectory`, only base
- it doesn't try to upload Xcode logs
- it doesn't return artifacts